packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -834,7 +837,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.reSubmit;
     }
 
+    public get disposeFn(): (error?: ICriticalContainerError) => void {
+        return this.context.disposeFn ?? this.context.closeFn;
+    }
+
     public get closeFn(): (error?: ICriticalContainerError) => void {
+        if (this.context.disposeFn) {
+            // Also call disposeFn to retain functionality of runtime being disposed on close
```

Comment:
Are there plans in the future to expose the ability to only close?

Reply:
I think there are some GC features that depend on the disposition on close, so those would also need to be adjusted. [AB#2953](https://dev.azure.com/fluidframework/internal/_workitems/edit/2953)